lib/src/rules/use_decorated_box.dart

```diff
@@ -91,20 +93,25 @@ class _Visitor extends SimpleAstVisitor {
 class _ArgumentData {
   var positionalArgumentsFound = false;
   var additionalArgumentsFound = false;
-  var hasDecoration = false;
+  var hasNonNullDecoration = false;
 
-  _ArgumentData(ArgumentList node) {
+  _ArgumentData(ArgumentList node, LinterContext context) {
     for (var argument in node.arguments) {
       if (argument is! NamedExpression) {
         positionalArgumentsFound = true;
         return;
       }
-      var label = argument.name.label;
-      if (label.name == 'decoration') {
-        hasDecoration = true;
-      } else if (label.name == 'child') {
+      var name = argument.name.label.name;
+      if (name == 'decoration') {
+        // Not lint if decoration is null or nullable because a DecoratedBox
+        // only accepts a non-null Decoration
+        if (DartTypeUtilities.isNonNullable(
+            context, argument.expression.staticType)) {
```

Comment:
Checking for nullability is good, but we're missing the more important test to ensure that the type of the expression is assignable to `BoxDecoration`.

If you wouldn't mind fixing that while you're at it, then also add a test to ensure that we don't lint when the `decoration` argument  is some other kind of decoration (maybe a `ShapeDecoration`).

Reply:
A `Container` accepts a `Decoration?`, which looks like a deliberate design decision to me. This also doesn't throw an error:

```dart
child: Container(decoration: const ShapeDecoration(shape: CircleBorder()))
```

If a Container is supposed to only accept a `BoxDecoration`, why not just allow the constructor to accept a `BoxDecoration` instead of a `Decoration`?